packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/undo/services/UndoRedoRecorder.java

```diff
@@ -40,6 +49,12 @@ private boolean canHandle(IInput input)  {
     public IInput preProcess(IEditingContext editingContext, IInput input, Sinks.Many<ChangeDescription> changeDescriptionSink) {
         if (editingContext instanceof EditingContext siriusEditingContext && canHandle(input)) {
             siriusEditingContext.getChangeRecorder().beginRecording(siriusEditingContext.getDomain().getResourceSet().getResources());
+            this.representationChangeEventRecorders.stream()
+                    .filter(recorder -> recorder.canHandle(editingContext, input))
+                    .forEach(recorder -> {
+                        var changes = recorder.getChanges(editingContext, input);
+                        siriusEditingContext.getRepresentationChangesDescription().put(input.id().toString(), changes);
```

Comment:
`siriusEditingContext.getRepresentationChangesDescription().computeIfAbsent(input.id().toString(), key -> new ArrayList<>()).addAll(changes);`

Reply:
What's the improvement here ? Seems to only create an additional list. You could do ` ArrayList<>(changes)` but I don't see why `computeIfAbsent`.
Note that later for diagrams we sometimes want to add changes to the existing list and not keeping or replacing the current list.

Reply:
I have done this change exactly because in the next PR what will be done is put the list of change if the list does not exist for a key.
So instead of doing :
```java
List<> listOfChanges = ...
if(map.get(key) != null) {
  map.get(key).addAll(listOfChanges)
} else {
  map.put(key, listOfChanges)
}
```
I initialize the map for the non-existing key with an empty list, then in all case doing `addAll(listOfChanges)`:
`map.computeIfAbsent(key, key -> new List()).addAll(listOfChanges)`

Reply:
done